docs/openapi.yml

```diff
@@ -278,6 +278,62 @@ paths:
               schema:
                 $ref: '#/components/schemas/datasets'
 
+  /datasources:
+    get:
+      summary: List all datasources
+      description: Returns a list of datasources.
+      tags:
+        - Datasources
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/datasources'
+    post:
+      summary: Create a datasource
+      description: "Creates a new datasource and returns the created datasource.
+      All datasource objects are immutable and are uniquely identified by their URN.
+        The name must also be globally unique.
+        The `connectionUrl` must consist of the following components separated by a colon:\n
+        * Access method (usually `jdbc:`)\n
+        * The datasource type, valid values of which are whitelisted in `DatasourceType.java`\n
+        * A host & port\n
+        * The scheme the scheme used"
+
+      tags:
+        - Datasources
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/createDatasource'
+      responses:
+        '201':
+          description: CREATED
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/datasources'
+
+  /datasources/{urn}:
+    parameters:
+      - $ref: '#/components/parameters/datasourceUrn'
+    get:
+      summary: Get a datasource
+      description: 'Returns a datasource response for the supplied urn.
```

Comment:
Let's simplify the wording for now:

**description**:  Retrieve a datasource.

We'll want to eventually introduce a section to the spec that describes the format of each URN and their usage.